SupportYourLocals.Data/DataStructures/UserData.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using MapControl;
+using SupportYourLocals.ExtensionMethods;
+
+namespace SupportYourLocals.Data
+{
+    public class UserData
+    {
+        public int ID { get; set; }
+        public string Username { get; set; }
+        public string PasswordHash { get; set; }
+        public string Salt { get; set; }
+
+        public UserData(string username, string passwordHash, string salt, int id)
+        {
+            ID = id;
```

Comment:
Could copy the ID generation function to this class, and use it analogously like in LocationData.